packages/compass-web/scripts/electron-proxy.js

```diff
@@ -0,0 +1,160 @@
+   */
+  static async getCloudSessionCookies() {
+    const cloudHostCookies = await session.defaultSession.cookies.get({
+      domain: CLOUD_HOST,
+    });
+    return cloudHostCookies.map((cookie) => {
+      return `${cookie.name}=${cookie.value}`;
+    });
+  }
+
+  /**
+   * @returns {Promise<{ groupId: string; }>}
+   */
+  static async authenticate() {
+    this.#authenticatePromise ??= (async () => {
+      await electronApp.whenReady();
+      const bw = new BrowserWindow({
+        height: 800,
+        width: 400,
+        resizable: false,
+        fullscreenable: false,
+      });
+      const abortController = new AbortController();
+      const authInfoPromise = Promise.race([
+        new Promise((resolve, reject) => {
+          /**
+           * @param {{ redirectURL: string }} details
+           */
+          async function waitForAuthentication({ redirectURL }) {
+            try {
+              if (!redirectURL.startsWith(`https://${CLOUD_HOST}/v2/`)) {
+                return;
+              }
+              const groupId = new URL(redirectURL).pathname.replace('/v2/', '');
+              resolve({ groupId });
+            } catch (err) {
+              reject(err);
+            }
+          }
+          bw.webContents.session.webRequest.onBeforeRedirect(
+            { urls: ['*://*/*'] },
+            waitForAuthentication
+          );
+        }).finally(() => {
+          queueMicrotask(() => {
+            abortController.abort();
+            electronApp.dock.hide();
+            bw.close();
+          });
+        }),
+        once(bw, 'close', { signal: abortController.signal }).then(() => {
+          throw new Error('Window closed before finished signing in');
+        }),
+      ]);
+      electronApp.dock.show();
+      void bw.loadURL(`https://${CLOUD_ACCOUNT_HOST}/account/login`);
+      return authInfoPromise;
+    })();
+    return this.#authenticatePromise;
+  }
+}
+
+// Proxy endpoint that triggers the sign in flow through configured cloud
+// environment
+proxyWebServer.use('/authenticate', async (req, res) => {
+  if (req.method !== 'POST') {
+    res.statusCode = 400;
+    res.end();
+    return;
+  }
+
+  try {
+    const { groupId } = await AtlasCloudAuthenticator.authenticate();
+    res.setHeader('Content-Type', 'application/json');
+    res.send(JSON.stringify({ groupId }));
+  } catch (err) {
+    res.statusCode = 500;
+    res.send(err.stack);
+    res.end();
+  }
+});
+
+// Prefixed proxy to the cloud backend
+proxyWebServer.use(
+  '/cloud-mongodb-com',
+  proxyMiddleware(`https://${CLOUD_HOST}`, {
+    async proxyReqOptDecorator(req) {
+      req.headers ??= {};
+      req.headers['host'] = CLOUD_HOST;
+      req.headers['origin'] = `https://${CLOUD_HOST}`;
+      req.headers['cookie'] =
+        await AtlasCloudAuthenticator.getCloudSessionCookies();
+      return req;
+    },
+  })
+);
+
+// Everything else is proxied directly to webpack-dev-server
+proxyWebServer.use(
```

Comment:
how come this is necessary? isn't this only acting as a proxy for the api?

Reply:
It's just more convenient for local dev than running all the commands separately and wiring stuff together manually, if you have a strong preference for this not to be here, I can remove this and write some instructions for how to run this locally without this instead

Reply:
how does it work then? is this serving the compass-web sandbox? The way I imagine this working is just:

```
npm start-cloud-backend-proxy 3001
npm run compass-web --cloud-backend=https://localhost:3001
```

is there more to it?

Reply:
Ahh is part of the webpack config on serve, got it